lib/segment/src/vector_storage/chunked_vectors.rs

```diff
@@ -32,7 +27,7 @@ impl<T: Copy + Clone + Default> ChunkedVectors<T> {
     pub fn new(dim: usize) -> Self {
         assert_ne!(dim, 0, "The vector's dimension cannot be 0");
         let vector_size = dim * mem::size_of::<T>();
-        let chunk_capacity = max(MIN_CHUNK_CAPACITY, CHUNK_SIZE / vector_size);
+        let chunk_capacity = CHUNK_SIZE / vector_size;
```

Comment:
This `min` covers case when `CHUNK_SIZE / vector_size` is `0`. How very  large DIM produces errors because of empty `chunk_capacity`. I propose to do the same for mmap chunks

Reply:
But we don't allow individual vectors to be larger than chunk. If we take validation into account, vectors are guaranteed to be smaller than chunk size

Reply:
Do we have DIM limit? I'm sure we have limit for multivectors. Also, we can add assert at least
```
assert_ne!(chunk_capacity, 0);
```

Reply:
> Do we have DIM limit?

yes, 65k